vignettes/caseStudy2-NutrientThreshold.Rmd

```diff
@@ -0,0 +1,443 @@
+(Reference: [Available energy fluxes drive a transition in the diversity, stability, and functional structure of microbial communities](https://doi.org/10.1371/journal.pcbi.1006793))
+
+The aim of this case study is to design and demonstrate the existence of nutrient concentration threshold which limits the beta-diversity of communities.
+
+To fulfill this aim, we designed a gradient of environments, as well as a gradient of communities.
+
+## Load dependencies
+
+```{r load, eval=TRUE}
+library(ggplot2)
+library(vegan)
+library(reshape2)
+library(umap)
+library(miaSim)
+library(philentropy)
+library(ape)
+library(cluster)
+
+```
+
+(This batch of simulations is time-consuming. To reduce the calculation burden, we decreased the numbers of environments, resources, and communities from the original 10 to 5, and made other minor modifications.)
+
+## Set random seed
+
+```{r}
+set.seed(42)
+```
+
+## initial shared parameters
+
+```{r}
+n_species <- 5
+n_resources <- 5
+E <- randomE(n_species, n_resources, mean_consumption = 1, mean_production = 3)
+growth_rates <- runif(n_species)
+monod_constant <- matrix(rbeta(n_species*n_resources, 10,10),nrow=n_species, ncol=n_resources)
+t_store <- 50
+n.instances <- 1 # no stochastic process: no need to repeat
+
+```
+
+## generating function
+this function generates a data frame, where each row is arranged in an increasing dissimilarity to the first row.
+
+```{r}
+gradient.df.generator <- function(n_row, n_col, density_row, max_gradient, error_interval){
+    list_initial <- list()
+    dissimilarity.gradient <- seq(from = 0, to = max_gradient, length.out = n_row)
+    for (i in seq_len(n_row)){
+        print(i)
+        if (i == 1){
+            row_temp <- rbeta(n_col, 1, 1/n_col)
+            col_to_remove <- sample(x = seq_len(n_col), size = n_col-n_col*density_row)
+            row_temp[col_to_remove] <- 0
+            list_initial[[i]] <- row_temp
+        } else {
+            while (length(list_initial) < i) {
+                row_temp <- rbeta(n_col, 1, 1/n_col)
+                col_to_remove <- sample(x = seq_len(n_col), size = n_col-n_col*density_row)
+                row_temp[col_to_remove] <- 0
+                diff_temp <- abs(vegdist(rbind(list_initial[[1]], row_temp), method = "bray") - dissimilarity.gradient[i])
+                if (diff_temp < error_interval) {
+                    list_initial[[i]] <- row_temp
+                }
+            }
+        }
+    }
+    dataframe_to_return <- as.data.frame(t(matrix(unlist(list_initial), ncol = n_row)))
+    return(dataframe_to_return)
+}
+```
+
+## generate communities
+
+```{r}
+n.community <- 5 # you can also try 20 or even 50.
+density.community <- 0.8
+set.seed(42)
+community.initial.df <- gradient.df.generator(n_row = n.community, n_col = n_species, density_row = density.community, max_gradient = 0.7, error_interval = 0.1)
+dist.community.initial.df <- vegdist(community.initial.df, method = "bray")
+
+```
+
+## load plotting functions
```

Comment:
This section and subsequent sections should be replaced with ready-made functionality in miaViz / @TuomasBorman could you check where possible?